MIPS/mip-0006-slot-reduction-90s.md

```diff
@@ -0,0 +1,270 @@
+
+We aim to deliver this MIP with a protocol upgrade scheduled for the Fall 2025.
+
+# Motivation
+
+Mina uses the Ouroboros Samasika consensus protocol, which divides time into equal-sized chunks called slots. Currently, the slot time is set at 3 minutes, which directly limits how frequently blocks in the Mina blockchain can be created, as consensus ensures at most one block per slot.
+
+The original slot time of 180 seconds was determined primarily by performance factors, including:
+
+1. The time required to create a Kimchi proof-system-based SNARK proof (5-15 seconds depending on hardware)
+2. The requirement that SNARK work for certain previous blocks must be completed before creating the current block
+3. Ensuring all nodes can receive blocks before the end of a slot
+4. Maximum number of account updates per zkapp transaction
+
+Recent protocol releases have dramatically improved block creation and transaction processing performance (see Appendix for details). These improvements now make it feasible to decrease the slot time without compromising system stability or performance.
+
+Reducing the slot time will allow for more frequent block creation, improving the network's throughput and user experience.
+
+# Specification
+
+Following protocol modifications are required to implement this change:
+
+1. Update the `block_window_duration_ms` constant from 180,000 to 90,000 milliseconds
+    - Effect: Slot time will be reduced from 3 minutes to 90 seconds
+2. Update the `coinbase` constant from 720 to 360
+    - Effect: Block creators will receive half of the current MINA amount for creating a block
+    - This maintains the same token emission rate as before, ensuring inflation is not affected by the slot time change
+3. Update the account vesting schedule logic
+    - As part of the Ledger Migration procedure (that will be executed by the Mina node as part of the upgrade), update vesting parameters for **actively vesting** accounts.
+    - For the definition of **actively vesting** accounts and formulas for update of vesting parameters, see Appendix.
+4. Remove the zkapp soft limit configuration option
+    - Currently set to 24 by default
+    - Not constrained by the protocol, but will be removed completely
+
+## Prerequisites
+
+This update requires a prerequisite improvement to the SNARK work production code, which will be delivered before the hard fork as part of a soft fork update. This improvement will allow coordinators to use workers more efficiently by giving them smaller pieces for proving. See Appendix for more details.
+
+After delivering that update, the network will require at least some of the community members running the SNARK workers to run multiple (at least 4) SNARK workers per coordinator to ensure timely processing of SNARK work for the heaviest transactions.
```

Comment:
Explain/clarify for readers the reasoning behind the number of snark workers per coordinator. Unclear how this was determined or the significance of it; or duration. It is expected to be a permanent increase? Does this have anything to do with the slot time reduction, or is it related to the prerequesite? If so, perhaps this is not relevant for this proposal?

Reply:
Can we also outline what is the specs here? when we talk for a snark workers as 128 core machine can be a single snark worker or be a 4 snark worker with 32 core allocated to each. It's very important for clarity and convey the exact compute requirements. 

Also what does constitutes as timely processing. Currently a transaction can sit on a pool for hours because of various reasons but mostly snark related delays.